### ISSUE TITLE
Tracking down instigators of SAVE

### ISSUE THREAD

**Issue by guybenron** [michaelcontento/redux-storage#100]
Love this package! :)

I'm using redux-storage-decorator-filter to restrict the save calls to a specific part of the state. I also just started using redux-sagas, so I put the three `EFFECT_*` actions into the storage blacklist, and they both work together beautifully.
However, I'm getting an extra `REDUX_STORAGE_SAVE` action which I have no clue why it's being dispatched, and there is no information I can see to help me figure it out. How do I track down the part of the code that caused the save operation to happen ... ?

Thanks !
-Guy

**Comment by michaelcontento** [comment]
You're welcome! :sunglasses: 

It should be pretty easy, as redux-storage simply dispatches a [whole new action](https://github.com/michaelcontento/redux-storage/blob/master/src/createMiddleware.js#L78-L80) (or in other words: no special dispatching magic involved!). So the action right before `REDUX_STORAGE_SAVE` should be the action that has triggerd this save operation. 

**NOTE**: If your using [redux-storage-decorator-debounce](https://github.com/michaelcontento/redux-storage-decorator-debounce) this relation is no longer valid, as it's the main job of `decorator-debounce` to delay the save operation :wink: 

Or is this the way you already tried to tackle it with no success?

**Comment by guybenron** [comment]
Thanks for responding :)

I've tried to track it down like you said, but redux-sagas uses actions very liberally (almost every API call results in two actions) so it makes tracking down the source of the save very confusing. What I can say for sure after following a single flow, is that the save action is dispatched even if the filtered state doesn't change.

I'm also using redux-logger to log all of the actions, but just to make sure I listened in to the redux-sagas actions and JSON.stringified the state to the console. I'm seeing a bunch of `EFFECT_TRIGGERED` and `EFFECT_RESOLVED`, but the state does not change :(
The only thing that comes to mind is that maybe the before state and after state, while identical, aren't the same object - immutability etc. But it's really grasping at straws for me at this point...

Any suggestions as to how to further break this down and hunt the errant code?
